scipy/differentiate/tests/test_differentiate.py

```diff
@@ -54,17 +55,19 @@ def test_vectorization(self, order, shape, xp):
         # input shapes.
         x = np.linspace(-0.05, 1.05, 12).reshape(shape) if shape else 0.6
         n = np.size(x)
+        state = {}
 
         @np.vectorize
         def _derivative_single(x):
             return derivative(self.f, x, order=order)
 
         def f(x, *args, **kwargs):
-            f.nit += 1
-            f.feval += 1 if (x.size == n or x.ndim <=1) else x.shape[-1]
+            state['nit'] += 1
+            state['feval'] += 1 if (x.size == n or x.ndim <=1) else x.shape[-1]
             return self.f(x, *args, **kwargs)
-        f.nit = -1
-        f.feval = 0
```

Comment:
Is there an automatic check now for this sort of thing to prevent this sort of thing in the future?

Reply:
There will be once we're done with all of this "make things thread-safe" set of PRs. Namely a CI run with [pytest-run-parallel](https://github.com/Quansight-Labs/pytest-run-parallel/).